test/endecode_test.c

```diff
@@ -79,15 +79,11 @@ static EVP_PKEY *make_template(const char *type, OSSL_PARAM *genparams)
     EVP_PKEY_CTX *ctx = NULL;
 
 # ifndef OPENSSL_NO_DH
-    /*
-     * Use 512-bit DH(X) keys with predetermined parameters for efficiency,
-     * for testing only. Use a minimum key size of 2048 for security purposes.
-     */
     if (strcmp(type, "DH") == 0)
-        return get_dh512(keyctx);
+        return get_dh2048(keyctx);
```

Comment:
Do we still care about the efficiency loss here?

Reply:
i will always advocate to use strong cryptography even in tests. Unless it is an explicit weak/old/compat test case.

To prevent us having issues where tests start to fail as they use obsolete key sizes and we lost ability to regenerate required test certs/keys/etc. Like we have stumbled upon during 3.4.0 development cycle.

Reply:
I would like @t8m 's opinion on this. I seem to remember him going thru and changing to weak keys in multiple places where I had larger keys beforehand.

Reply:
In some cases using non-weak keys can be really prohibitive in terms of CPU consumption for tests. Hopefully this is not the case here.